### ISSUE TITLE
Lights not removed or added to scenes after room is created

### ISSUE THREAD

**Issue by alexyao2015** [diyhue/diyHue#178]
**Describe the bug**
Scenes to not update when lights are updated in a room.

**To Reproduce**
Method 1: Add two lights to a room. Remove one. Tap a default scene. Both lights change even though only one light is in the room.
Method 2: Add a light to an existing room. Tap a scene. 

**Expected behavior**
Only lights in room should change when selecting scenes.

**Comment by alexyao2015** [comment]
This is still an issue and needs to be investigated.

**Comment by cheesemarathon** [comment]
Can you post the contents of your config.json file

**Comment by alexyao2015** [comment]
Alright so it seems it does add bulbs to savanna sunset, tropical twilight, arctic aurora, and spring blossom. The bulb is not added to the other scenes. 

Removing the bulb from the room does not remove it out of any scenes.

Also a scene is not removed from the config when a room is removed (1 bulb room, that bulb is added to a different room, entire room is gone but scenes are left). 

Except from config: 5 and 1 were added after the room was made. 
```
    "scenes": {
        "1": {
            "appdata": {
                "data": "2oAJy_r01_d15",
                "version": 1
            },
            "group": "1",
            "lastupdated": "2019-09-07T19:41:29",
            "lightstates": {
                "1": {
                    "bri": 198,
                    "on": true,
                    "xy": [
                        0.533664,
                        0.386149
                    ]
                },
                "2": {
                    "bri": 198,
                    "on": true,
                    "xy": [
                        0.533664,
                        0.386149
                    ]
                },
                "3": {
                    "bri": 198,
                    "on": true,
                    "xy": [
                        0.463189,
                        0.450516
                    ]
                },
                "4": {
                    "bri": 199,
                    "on": true,
                    "xy": [
                        0.532712,
                        0.386129
                    ]
                },
                "5": {
                    "bri": 198,
                    "on": true,
                    "xy": [
                        0.588323,
                        0.353963
                    ]
                }
            },
            "locked": false,
            "name": "Savanna sunset",
            "owner": "a7161538be80d40b3de98dece6e91f90",
            "picture": "",
            "recycle": false,
            "type": "GroupScene",
            "version": 2
        },
        "10": {
            "appdata": {
                "data": "BqYOX_r01_d06",
                "version": 1
            },
            "group": "1",
            "lastupdated": "2019-09-07T19:41:32",
            "lightstates": {
                "2": {
                    "bri": 77,
                    "ct": 367,
                    "on": true
                },
                "3": {
                    "bri": 77,
                    "ct": 367,
                    "on": true
                },
                "4": {
                    "bri": 77,
                    "ct": 367,
                    "on": true
                }
            },
            "locked": false,
            "name": "Dimmed",
            "owner": "a7161538be80d40b3de98dece6e91f90",
            "picture": "",
            "recycle": false,
            "type": "GroupScene",
            "version": 2
        },
```

**Comment by mariusmotea** [comment]
Lights must be removed from scenes once they are deleted. Code is in place and tested. Maybe with you config some errors where generated:

```
               # Delete the light from any scenes
                for scene in list(bridge_config["scenes"]):
                    if "lights" in bridge_config["scenes"][scene] and del_light in bridge_config["scenes"][scene]["lights"]:
                        bridge_config["scenes"][scene]["lights"].remove(del_light)
                        del bridge_config["scenes"][scene]["lightstates"][del_light]
                        if len(bridge_config["scenes"][scene]["lights"]) == 0:
                            del bridge_config["scenes"][scene]```

**Comment by alexyao2015** [comment]
Yes, that piece of code works fine. There needs to be a other loop to add/remove lights when the scene is updated

**Comment by alexyao2015** [comment]
No

**Comment by alexyao2015** [comment]
No

**Comment by cheesemarathon** [comment]
@mariusmotea I think we have found the general issue here. There is a loop missing from the section of code above to delete lights when the scene is updated. Is this the case? Can this be done?

**Comment by cheesemarathon** [comment]
This won't be closed by stale bot now. It's clearly an important issue to fix

**Comment by mariusmotea** [comment]
I made some changes. Seams the scene can be not only lights based but also group based. In both cases the lights are listed in "lightstates" so i changed the loop for this key. I also made the scene to be deleted in case the lightates key has no lights.

**Comment by alexyao2015** [comment]
I don't think that fixed it. I still have it where lights change in a scene when they are not part of the actual room. Maybe create a loop that checks if the light is in a group and remove/add it if it is?

**Comment by alexyao2015** [comment]
I think diyhue is ignoring the group that the scene is associated with. It needs to check to make sure the lights are part of the group and then set the light state if it is.

**Comment by mariusmotea** [comment]
I believe the issue triggered when the lights are migrated from one room to another. Somehow i need to find a way to trigger a function when this is happening.

**Comment by alexyao2015** [comment]
Ok I believe I've found the problem and know how to fix it, just need to find time to do so. Basically, the lightstates are being used now to set all lights that have a lightstate to the scene. This is incorrect.

A light can have a lightstate but should not actually be changed unless the light is in the lights key. For rooms, this key should be dynamically generated by diyhue (not stored in the config) by grabbing the lights from the room. 

Additionally, when requesting scenes, this should not show lightstates at all (it currently does). It should only output lights (dynamically generated). Lightstates are only ever shown when `/api/<username>/scenes/<id>` is called. 

This potentially may fix #19 since the issue there is scene related. Likely related to incorrect scene output as seen here. 

All information here was gathered from the hue api docs.

**Comment by alexyao2015** [comment]
Ok I ran additional tests and have confirmed that this is indeed the problem with #19. Fixing this will resolve this problem as well as #19.

**Comment by alexyao2015** [comment]
I've created a partial fix for setting lights. Still have create a loop to delete scenes when a room/zone is deleted.
